gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -174,6 +175,16 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 	if err != nil {
 		return err
 	}
+
+	// enforce sorting msg.Files based on Go conventions for predictability
+	// https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/sort/sort.go;l=33
```

Comment:
Is this link necessary?  I think it should be straightforward enough for someone to find the sort package documentation if they aren't familiar.

Reply:
From what I can find, Go parses files _lexically_ and returns the resulting AST. As for building, perhaps more research is needed to look into the action graph for Go, but from my research this dependency tree's order is alphabetical, as shown [here (build.ReadDir)](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/go/build/build.go;l=106) and [here (os.ReadDir)](https://cs.opensource.google/go/go/+/refs/tags/go1.16beta1:src/os/dir.go;l=111).

[This](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/sort/sort.go;l=21-23) bit about the comparison operation is interesting and could prove to be an issue, so we may need to replace with a more stable sort variant instead to ensure order remains consistent in all cases. Having a test or two written may address the concern, though I wasn't sure how best to go about that.

> 	// If both Less(i, j) and Less(j, i) are false,
	// then the elements at index i and j are considered equal.
        // Sort may place equal elements in any order in the final result